project/WeaverPlugin.scala

```diff
@@ -429,14 +429,15 @@ object WeaverPlugin extends AutoPlugin {
           var projectId = lp.project
 
           val scalaAxis =
-            if (projectId.endsWith(scala3Suffix)) {
+            if (projectId.endsWith(scala3Suffix) && !projectId.endsWith(ce3Suffix)) {
```

Comment:
This is ugly and horrible, and commandmatrix was going to make this better, until I realise there's some weird non-lazy initialisation going on somewhere :D 

So it will have to wait.